### PR TITLE
Add support for .ico files

### PR DESCRIPTION
**Pull request by leolabs**
This PR adds support for .ico and .cur files, which both use the same file structure.

When a given .ico/.cur file contains multiple images, then the largest image's size is returned.

### CODE REVIEW

File: lib/parse_stream/ico.js
Comment (puzrin) [inline lib/parse_stream/ico.js:25]: missed `parser._skipBytes(Infinity);`

File: lib/parse_stream/ico.js
Comment (puzrin) [reply lib/parse_stream/ico.js:20]: missed fail if `numImages` === 0;

**Review by puzrin — CHANGES_REQUESTED** [review]
Look reasonable, but see comments.

Also please rebase & squash commits.

File: lib/parse_stream/ico.js
Comment (puzrin) [inline lib/parse_stream/ico.js:19]: Here still missed `parser._skipBytes(Infinity);`

When you stop analyse content, the rest should be sent to /dev/null. Or you can block other (parallel) parsers.

**Comment by leolabs** [comment]
Thanks for your quick review! I've made some changes based on your comments and the PR should now be ready to be merged. You can squash all commits into one during the merge using the menu next to the merge button:

Screenshot 2020-10-29 at 22 14 25

File: package.json
Comment (puzrin) [inline package.json:11]: remove cur?

**Comment by puzrin** [comment]
> You can squash all commits into one during the merge using the menu next to the merge button:

I know, but i'd like to see CI results before merge. Because of suspisious tests failure https://travis-ci.org/github/nodeca/probe-image-size/jobs/739931912. I also disabled one tricky test in master. That's why i asked to rebase.

--

In general, everything should be fine now.

**Comment by leolabs** [comment]
Ah, alright. I've rebased it :)

**Review by puzrin — APPROVED** [review]

**Comment by puzrin** [comment]
`npm run coverage` - see report.

Could you improve tests coverage for your code? See jpeg tests for example, how to write test vectors with ease (instead of real files).

**Comment by puzrin** [comment]
See https://github.com/nodeca/probe-image-size/blob/master/test/formats.js#L194

Since it's not convenient tot create real files for all edge cases, we create fake data instead.

**Pull request merged**
